graylog2-server/src/main/java/org/graylog2/indexer/cluster/Cluster.java

```diff
@@ -130,28 +136,48 @@ public JsonArray catNodes(String fields) {
     }
 
     public String nodeIdToName(String nodeId) {
-        final JsonObject nodeInfo = getNodeInfo(nodeId);
-        return nodeInfo == null ? "UNKNOWN" : nodeInfo.getAsJsonPrimitive("name").getAsString();
-
+        return getNodeInfo(nodeId)
+                .map(nodeInfo -> nodeInfo.get("name"))
+                .filter(JsonElement::isJsonPrimitive)
+                .map(JsonElement::getAsJsonPrimitive)
+                .filter(JsonPrimitive::isString)
+                .map(JsonPrimitive::getAsString)
+                .orElse("UNKNOWN");
     }
 
     public String nodeIdToHostName(String nodeId) {
-        final JsonObject nodeInfo = getNodeInfo(nodeId);
-        return nodeInfo == null ? "UNKNOWN" : nodeInfo.getAsJsonPrimitive("host").getAsString();
+        return getNodeInfo(nodeId)
+                .map(nodeInfo -> nodeInfo.get("host"))
+                .filter(JsonElement::isJsonPrimitive)
+                .map(JsonElement::getAsJsonPrimitive)
+                .filter(JsonPrimitive::isString)
+                .map(JsonPrimitive::getAsString)
+                .orElse("UNKNOWN");
```

Comment:
See above, plus: "UNKNOWN" is a valid host name. If someone ever wants to actually use the result of this beyond displaying it, that could be rather nasty.